### PR TITLE
Enforce rate-limit for backend-login

### PR DESCRIPTION
**Pull request by ymo-sci**
### Description

it seems, that there was rate-limiting years ago, but got removed in https://github.com/scireum/sirius-biz/commit/613257814#diff-11ffed5b48f839e6d599427f3f00a4510ef156fd08267b2fb8b92c1ac138cebeL48-L49 the security-block is now adjusted to only allow 10 wrong logins in 2 minutes, then the rate-limit will stop the user earlier for trying more users/passwords

Bildschirmfoto 2026-05-27 um 16 04 51

- after the 4th bad login a Captcha is included now in the login-form which should be ok for real users with typos

### Additional Notes

- This PR fixes or works on following ticket(s): [SIRI-1217](https://scireum.myjetbrains.com/youtrack/issue/SIRI-1217)
- This PR is related to PR: 

### Checklist

- [x] Code change has been tested and works locally
- [x] Code was formatted via IntelliJ and follows SonarLint & [best practices](https://scireum.myjetbrains.com/youtrack/articles/MISC-A-16/CodeStyle-JavaDoc)
- [ ] Patch Tasks: Is local execution of Patch Tasks necessary? If so, please also mark the PR with the tag.

### CODE REVIEW

**Review by sabieber — APPROVED** [review]
> BREAKING: to use it safe, http.captcha.secret = "secret" must be filled per system - a default is provided.

We got security feedback for other unsafe defaults before so should we really provide an unsafe default here or rather produce an error when the secret is not set when trying to be used?

File: src/test/resources/test.conf [hunk 1]
```diff
@@ -102,6 +102,12 @@ isenguard.limit.test {
     limit = 5
 }
 
+isenguard.limit.security {
+    interval = 1m
+    limit = 3
+    type = "ip"
```
Comment (idlira) [inline src/test/resources/test.conf:108]: Might be a bit edge case but several users in a company will land on our server under the same IP. So, 4 attempts **per user** is not really guaranteed unless we also include the user name in the scope.

Comment (ymo-sci) [reply src/test/resources/test.conf:108]: this is the test.conf - for prod a limit of 10 / 2m would be used and only for bad logins - think that should never be the case...

Comment (idlira) [reply src/test/resources/test.conf:108]: It's not about being enough or not. It's about the information being displayed to the user being wrong. As soon as 2 people make a mistake, one is stealing the limit from the other. And all you have to do is to setup the limit having the user name with the source IP concatenated.

Comment (sabieber) [reply src/test/resources/test.conf:108]: Mhhh I thought about the concatenation idea a bit and I think this could mean an attacker could optimize for it if he finds that out. As in: he could simply iterate over the accounts he wants to breach and make an attempt for each, then rinse and repeat.

Comment (idlira) [reply src/test/resources/test.conf:108]: Well, we can leave as is. The goal was to protect for non intentional password change attempts. Users in regular use shall not even feel the difference. But the captcha logic would also trigger a bit sooner in my previous example.

Comment (ymo-sci) [reply src/test/resources/test.conf:108]: > As soon as 2 people make a mistake, one is stealing the limit from the other.

Yea ok, but only in the configured 2 minutes... i think bad luck for this edge case

Comment (idlira) [reply src/test/resources/test.conf:108]: > he could simply iterate over the accounts he wants to breach and make an attempt for each, then rinse and repeat.

well, than we would need 2 limits: one for each user name (rate limit + CAPTCHA) and additionally one for IP trying to cover the attacker. Again, we can think about it in a later step and go for it with this version.

**Review by bkositza — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by jakobvogel — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Comment by ymo-sci** [comment]
> > BREAKING: to use it safe, http.captcha.secret = "secret" must be filled per system - a default is provided.
> 
> We got security feedback for other unsafe defaults before so should we really provide an unsafe default here or rather produce an error when the secret is not set when trying to be used?

https://scireum.myjetbrains.com/youtrack/issue/SIRI-1225
